scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -380,10 +380,9 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
    * @tparam U The type of representative values used to dedup.
    * @group transform
    */
-  def distinctBy[U](f: T => U)(implicit ctu: ClassTag[U]): SCollection[T] =
-    this.pApply(Distinct
-      .withRepresentativeValueFn(Functions.serializableFn(f))
-      .withRepresentativeType(TypeDescriptor.of(ctu.runtimeClass).asInstanceOf[TypeDescriptor[U]]))
+  // This is simplier than Distinct.withRepresentativeValueFn, and allows us to set Coders
+  def distinctBy[U](f: T => U)(implicit toder: Coder[T], uoder: Coder[U]): SCollection[T] =
+    this.transform(me => me.groupBy(f).values.map(_.head))
```

Comment:
@regadas wouldn't this be less performant than the old implementation? Right now no combination will be applied...

Reply:
Hey @martinbomio, thanks for pointing this out.
I think the following implementation should work better for cases where one key might have values that doesn't fit in memory. Though I am not sure if the Iterable[V] returned by the groupBy would be materialised in memory.

```
  def distinctBy[U](f: T => U)(implicit toder: Coder[T], uoder: Coder[U]): SCollection[T] =
    this.transform { me =>
      me.keyBy(f).combineByKey(identity) { case (c, _) => c } { case (c, _) => c }.values
    }
```

What do you think? I'll create a fix with this.

Reply:
Good point @martinbomio I missed that! Thanks! I need to have a better look at this